docs/rules/define-emits-declaration.md

```diff
@@ -0,0 +1,67 @@
+---
+pageClass: rule-details
+sidebarDepth: 0
+title: vue/define-emits-declaration
+description: enforce declaration style of `defineEmits`
+---
+# vue/define-emits-declaration
+
+> enforce declaration style of `defineEmits`
+
+- :exclamation: <badge text="This rule has not been released yet." vertical="middle" type="error"> ***This rule has not been released yet.*** </badge>
+
+## :book: Rule Details
+
+This rule enforces `defineEmits` typing style which you should use `type-based` or `runtime` declaration.
+
+This rule only works in setup script and `lang="ts"`.
+
+<eslint-code-block :rules="{'vue/define-emits-declaration': ['error']}">
+
+```vue
+<script setup lang="ts">
+/* ✓ GOOD */
+const emit = defineEmits<{
+  (e: 'change', id: number): void
+  (e: 'update', value: string): void
+}>()
+</script>
+```
+
+</eslint-code-block>
+
+<eslint-code-block :rules="{'vue/define-emits-declaration': ['error']}">
+
+```vue
+<script setup lang="ts">
+/* ✗ BAD */
+const emit = defineEmits(['change', 'update'])
+</script>
+```
+
+</eslint-code-block>
+
+## :wrench: Options
+
+- `type-based` (default) enforces type-based declaration
+- `runtime` enforces runtime declaration
+
+```json
+  "vue/define-emits-declaration": ["error", "type-based" | "runtime"]
```

Comment:
Please also add an example with code block that explains what happens when the option is set to `runtime` (for the other rule as well).

See e.g. https://eslint.vuejs.org/rules/v-for-delimiter-style.html#options